<launch>
    <arg name="arm_command_gui" default="false"/>
    <!-- Note that changing these from the default requires matching changes in the Java code -->
    <arg name="left_arm_recv_url" default="udp://10.10.10.169:30002"/>
    <arg name="right_arm_recv_url" default="udp://10.10.10.169:30001"/>

    <arg name="model" default="$(find-pkg-share victor_description)/urdf/victor.urdf.xacro"/>

    <node pkg="victor_hardware" exec="arm_command_gui.py" name="arm_command_gui" if="$(var arm_command_gui)" namespace="victor" output="screen"/>
    <node pkg="victor_hardware" exec="iiwa_lcm_bridge_node" name="left_arm_lcm_bridge_node" output="screen" namespace="victor/left_arm">
        <param name="cartesian_pose_frame" value="victor_left_arm_world_frame_kuka"/>
        <param name="send_lcm_url" value="udp://10.10.10.12:30000"/>
        <param name="recv_lcm_url" value="$(var left_arm_recv_url)"/>
    </node>
    <node pkg="victor_hardware" exec="iiwa_lcm_bridge_node" name="right_arm_lcm_bridge_node" output="screen" namespace="victor/right_arm">
        <param name="cartesian_pose_frame" value="victor_right_arm_world_frame_kuka"/>
        <param name="send_lcm_url" value="udp://10.10.10.11:30000"/>
        <param name="recv_lcm_url" value="$(var right_arm_recv_url)"/>
    </node>

    <node pkg="joy" exec="joy_node" name="xbox_joystick" namespace="victor"/>
    <node pkg="victor_hardware" exec="robotiq_grippers_joystick_node.py" name="robotiq_grippers_joystick_node" namespace="victor">
        <param name="enable_finger_open_close_control" value="true"/>
        <param name="enable_scissor_open_close_control" value="true"/>
    </node>
    <node pkg="victor_hardware" exec="arm_wrench_republisher.py" name="arm_wrench_republisher"
          output="screen" namespace="victor"/>
    <!-- <node pkg="victor_hardware" exec="grasp_status_node.py" name="grasp_status"/> -->

</launch>
